extensions/container-image/container-image-s2i/deployment/src/main/java/io/quarkus/container/image/s2i/deployment/S2iBaseJavaImage.java

```diff
@@ -0,0 +1,64 @@
+
+package io.quarkus.container.image.s2i.deployment;
+
+import java.util.Optional;
+
+import io.quarkus.container.image.deployment.util.ImageUtil;
+
+public enum S2iBaseJavaImage {
+
+    FABRIC8("fabric8/s2i-java:2.3", "JAVA_MAIN_CLASS", "JAVA_APP_JAR", "JAVA_LIB_DIR", "JAVA_CLASSPATH", "JAVA_OPTIONS"),
+    OPENJDK("registry.access.redhat.com/openjdk/openjdk-11-rhel7", "JAVA_MAIN_CLASS", "JAVA_APP_JAR", "JAVA_LIB_DIR",
```

Comment:
There are 4 Red Hat OpenJDK images that I'm aware of (OpenJDK 8 and 11, RHEL 7 and 8), some of them being available from 2 registries (authenticated and unauthenticated).

- OpenJDK 8 (I'm not sure if we still support Java 8, but we probably do. Defaulting to 11 is probably good though.)
    - RHEL 7
        - registry.access.redhat.com/redhat-openjdk-18/openjdk18-openshift
        - registry.redhat.io/redhat-openjdk-18/openjdk18-openshift
    - RHEL 8
        - registry.redhat.io/openjdk/openjdk-8-rhel8 (registry.access.redhat.com variant not available)
- OpenJDK 11
    - RHEL 7
        - registry.access.redhat.com/openjdk/openjdk-11-rhel7
        - registry.redhat.io/openjdk/openjdk-11-rhel7
    - RHEL 8
        - registry.redhat.io/openjdk/openjdk-11-rhel8 (registry.access.redhat.com variant not available)

They all have different tags (as does the Fabric8 image), which should all be recognized. (That is, tag should be ignored when searching.)

Reply:
Added more images. Also cleaned them up a little bit, as the registry is stripped when performing the comparison.